medusa/providers/torrent/html/beyondhd.py

```diff
@@ -0,0 +1,189 @@
+# coding=utf-8
+
+"""Provider code for Beyond-hd."""
+
+from __future__ import unicode_literals
+
+import logging
+
+from medusa import tv
+from medusa.bs4_parser import BS4Parser
+from medusa.helper.common import convert_size
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+from requests.compat import urljoin
+from requests.utils import dict_from_cookiejar
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class BeyondHDProvider(TorrentProvider):
+    """Beyond-hd Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(BeyondHDProvider, self).__init__('Beyond-HD')
+
+        self.username = None
+        self.password = None
+
+        self.url = 'https://beyond-hd.me'
+        self.urls = {
+            'login': urljoin(self.url, 'login'),
+            'search': urljoin(self.url, 'torrents'),
+        }
+
+        # Proper Strings
+        self.proper_strings = ['PROPER', 'REPACK', 'REAL', 'RERIP']
+
+        # Miscellaneous Options
+        self.freeleech = True
```

Comment:
The freeleech option is not needed I think. There is nothing in the code that actually filters freeleech results.